data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/metrics/Metrics.java

```diff
@@ -175,9 +179,13 @@ public static <K, V> AutoCloseable register(final Consumer<K, V> consumer) {
    * @return A meter binder to close once the producer is closed.
    */
   public static <K, V> AutoCloseable register(final Producer<K, V> producer) {
-    final var clientMetrics = new KafkaClientMetrics(producer);
-    clientMetrics.bindTo(getRegistry());
-    return clientMetrics;
+    final var registry = getRegistry();
+    if (registry != null) {
```

Comment:
why was it null? looking at the old `bindTo`?

Reply:
I tested the option of disabling metrics, so it might be null in that case.